src/CoreWCF.AzureQueueStorage/src/CoreWCF/Channels/DeadLetterQueue.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Storage.Queues;
+using Azure.Storage.Queues.Models;
+
+namespace CoreWCF.Channels
+{
+    internal class DeadLetterQueue : IQueueBase
+    {
+        private QueueClient _client;
+        private ArraySegment<byte> _messageBuffer;
+
+        public DeadLetterQueue(string connectionString, string queueName)
+        {
+            _client = new QueueClient(connectionString, queueName);
+        }
+
+        public QueueClient queueClient { get => _client; set => _client = value; }
+
+        public Response DeleteMessage(string messageId, string popReceipt, CancellationToken cancellationToken = default)
+        {
+            return _client.DeleteMessage(messageId, popReceipt, cancellationToken);
+        }
+
+        public Task<Response<QueueMessage>> ReceiveMessageAsync(TimeSpan? visibilityTimeout = null, CancellationToken cancellationToken = default)
+        {
+            return _client.ReceiveMessageAsync(visibilityTimeout, cancellationToken);
+        }
+
+        public async Task SendMessageAsync(Message requestMessage)
+        {
```

Comment:
I don't think this method is being used as you called it in AzureQueueStorageQueueTransport.NotifyError. I presume this class would be used by multiple requests at once, which would mean a shared message buffer would be a bad idea.

Reply:
yeah thats a good point. moving the buffer inside the method.

Reply:
_messageBuffer isn't getting populated with any data. You need to either call the message encode to encode it, or have access to the original request data